fetch.js

```diff
@@ -93,60 +112,15 @@
   }
 
   Headers.prototype.keys = function() {
-    var items = []
-    this.forEach(function(value, name) { items.push(name) })
-    var iterator = {
-      next: function() {
-        var value = items.shift()
-        return {done: value === undefined, value: value}
-      }
-    }
-
-    if (support.iterable) {
-      iterator[Symbol.iterator] = function() {
-        return iterator
-      }
-    }
-
-    return iterator
+    return getIterator(this, function(value, name) { this.push(name) })
```

Comment:
Cleanup looks good. It's not immediately clear what `this` inside the iteration callback refers to, but that is cleared up when you look up `getIterator` source.